### PR TITLE
Fix unwind info in crypto/rc4/asm/rc4-x86_64.pl

### PR DESCRIPTION
**Pull request by hjl-tools**
Move .cfi_startproc to the right place for RC4.  Add missing
.cfi_startproc and .cfi_endproc to RC4_options.

### CODE REVIEW

**Comment by slontis** [comment]
Why is there no .cfi_endproc after the ret at line 541?

**Comment by hjl-tools** [comment]
> Why is there no .cfi_endproc after the ret at line 541?

.cfi_endproc  is placed at the end of function, not after ret:

```
.globl  RC4_options
.type   RC4_options,\@abi-omnipotent
.align  16
RC4_options:
.cfi_startproc
        lea     .Lopts(%rip),%rax
        mov     OPENSSL_ia32cap_P(%rip),%edx
        bt      \$20,%edx
        jc      .L8xchar
        bt      \$30,%edx
        jnc     .Ldone
        add     \$25,%rax
        ret
.L8xchar:
        add     \$12,%rax
.Ldone:
        ret
.cfi_endproc
.align  64
.Lopts:
.asciz  "rc4(8x,int)"
.asciz  "rc4(8x,char)"
.asciz  "rc4(16x,int)"
.asciz  "RC4 for x86_64, CRYPTOGAMS by <appro\@openssl.org>"
.align  64
.size   RC4_options,.-RC4_options
```

File: crypto/rc4/asm/rc4-x86_64.pl [hunk 1]
```diff
@@ -541,6 +543,7 @@ sub RC4_loop {
 	add	\$12,%rax
 .Ldone:
 	ret
+.cfi_endproc
```
Comment (bernd-edlinger) [inline crypto/rc4/asm/rc4-x86_64.pl]: Sorry, I am late at the party, but could you add a .size RC4_options,.-RC4_options here as well?

Comment (hjl-tools) [reply crypto/rc4/asm/rc4-x86_64.pl]: There are

```
globl  RC4_options
.type   RC4_options,\@abi-omnipotent
.align  16
RC4_options:
        lea     .Lopts(%rip),%rax
        mov     OPENSSL_ia32cap_P(%rip),%edx
        bt      \$20,%edx
        jc      .L8xchar
        bt      \$30,%edx
        jnc     .Ldone
        add     \$25,%rax
        ret
.L8xchar:
        add     \$12,%rax
.Ldone:
        ret
.align  64
.Lopts:
.asciz  "rc4(8x,int)"
.asciz  "rc4(8x,char)"
.asciz  "rc4(16x,int)"
.asciz  "RC4 for x86_64, CRYPTOGAMS by <appro\@openssl.org>"
.align  64
.size   RC4_options,.-RC4_options
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment (bernd-edlinger) [reply crypto/rc4/asm/rc4-x86_64.pl]: ah, ok, that's probably good enough then.

**Comment by bernd-edlinger** [comment]
Hmm, I forgot to ask, if your approvals are also good for 1.1.1 ?

**Comment by levitte** [comment]
I agree with cherry picking to 1.1.1

**Comment by bernd-edlinger** [comment]
Okay, merged to master as 967ef73, and 1.1.1 as d6116a9.
Thanks everybody!

**Pull request closed without merging**